weblab/static/js/entity.js

```diff
@@ -750,6 +751,40 @@ function init() {
     }
   }
 
+  /**
+   * Asynchronously submit a list of experiments to the front-end for execution on the task queue.
+   * Will register a callback every 0.2s to submit the next experiment until they're all done,
+   * so the UI doesn't freeze.
+   * @param plannedExperiments  array of objects with keys model, model_version, protocol, protocol_version
+   *     specifying the experiments to launch. May be empty.
+   * @param submitNow  whether to submit a job now, or just register the first callback.
+   */
+  function triggerExperimentSubmission(plannedExperiments, submitNow)
+  {
+    if (plannedExperiments && submitNow)
+    {
+      // Submit the first planned experiment to the queue
+      var exptSpec = plannedExperiments.pop();
+      exptSpec["task"] = "newExperiment";
```

Comment:
This raises an error on the last iteration because `plannedExperiments` is an empty array so `pop()` returns undefined. The`if` statement needs a `.length` test on the array as boolean testing an array won't have the desired effect.

Reply:
Yeah, I fixed that in #116 when I noticed it. Should really pull the fix into this PR, but...

Reply:
Fair enough :) I'll take a look at that one in a minute.